File: xcode_wkwebview_swift/testWebViewMac/ViewController.swift
//
//  ViewController.swift
//  testWebViewMac
//
//  Created by Alexey Strakh on 6/1/22.
//

import Cocoa
import WebKit

class ViewController: NSViewController, NSWindowDelegate, WKUIDelegate {

    var webView: WKWebView!

    override func viewDidLoad() {
        super.viewDidLoad()
        self.initWebView();
    }
    
    func initWebView() {
        let webConfiguration = WKWebViewConfiguration()
        webView = WKWebView(frame: .zero, configuration: webConfiguration)
        webView.uiDelegate = self
        self.view.addSubview(webView)
        let myURL = URL(string:"https://www.apple.com")
        let myRequest = URLRequest(url: myURL!)
        webView.load(myRequest)
    }

    override var representedObject: Any? {
        didSet {
        // Update the view, if already loaded.
        }
    }
    
    func windowDidResize(_ notification: Notification) {
        webView.frame = view.frame
    }
    
    override func viewWillAppear() {
        view.window?.delegate = self
    }
}

File: dotnet_wkwebview_cocoa/macWebViewTestCocoa/ViewController.cs
using System;

using AppKit;
using CoreGraphics;
using Foundation;
using WebKit;

namespace macWebViewTestCocoa
{
    public partial class ViewController : NSViewController, INSWindowDelegate, IWKUIDelegate
    {
        private WKWebView _webView;

        public ViewController (IntPtr handle) : base (handle)
        {
        }

        public override void ViewDidLoad ()
        {
            base.ViewDidLoad ();
            this.InitWebView();
            // Do any additional setup after loading the view.
        }

        private void InitWebView()
        {
            var webConfiguration = new WKWebViewConfiguration();
            webConfiguration.Preferences.SetValueForKey(NSObject.FromObject(true), new NSString("developerExtrasEnabled"));
            _webView = new WKWebView(frame: new CGRect(0,0,640,480), configuration: webConfiguration);
            _webView.UIDelegate = this;
            this.View.AddSubview(_webView);
            var myURL = new NSUrl("https://www.apple.com");
            var myRequest = new NSUrlRequest(url: myURL);
            _webView.LoadRequest(myRequest);
        }

        public override NSObject RepresentedObject {
            get {
                return base.RepresentedObject;
            }
            set {
                base.RepresentedObject = value;
                // Update the view, if already loaded.
            }
        }

    //    func windowDidResize(_ notification: Notification)
    //    {
    //        webView.frame = view.frame
    //}

        public override void ViewWillAppear()
        {
            if (this.View.Window != null)
                this.View.Window.Delegate = this;

            base.ViewWillAppear();
        }
    }
}
